src/gdal/ogr_file_format.cpp

```diff
@@ -1408,5 +1383,593 @@ LatLon OgrFileImport::calcAverageLatLon(OGRDataSourceH data_source)
+
+		auto pt = ogr::unique_geometry(OGR_G_CreateGeometry(wkbPoint));
+		QPointF proj_cord = georef.toProjectedCoords(object->asPoint()->getCoordF());
+
+		OGR_G_SetPoint_2D(pt.get(), 0, proj_cord.x(), proj_cord.y());
+
+		if (need_wgs1984)
+			OGR_G_Transform(pt.get(), transformation.get());
+
+		OGR_F_SetGeometry(po_feature.get(), pt.get());
+
+		OGR_F_SetStyleString(po_feature.get(),
+		                     OGR_STBL_Find(table.get(), QString::number(map->findSymbolIndex(symbol)).toLatin1().constData()));
+
+		if (OGR_L_CreateFeature(layer, po_feature.get()) != OGRERR_NONE)
+			throw FileFormatException(tr("Failed to create point feature in layer"));
+	};
+
+	map->applyOnMatchingObjects(add_feature, condition);
+}
+
+void OgrFileExport::addTextToLayer(OGRLayerH layer, const std::function<bool (const Object*)>& condition)
+{
+	const auto& georef = map->getGeoreferencing();
+
+	auto add_feature = [&](Object* object) {
+		auto po_feature = ogr::unique_feature(OGR_F_Create(OGR_L_GetLayerDefn(layer)));
+
+		QString sym_name = object->asText()->getText();
+		sym_name.truncate(32);
+		OGR_F_SetFieldString(po_feature.get(), OGR_F_GetFieldIndex(po_feature.get(), "Name"), sym_name.toLatin1().constData());
+
+		auto pt = ogr::unique_geometry(OGR_G_CreateGeometry(wkbPoint));
+		QPointF proj_cord = georef.toProjectedCoords(object->asText()->getAnchorCoordF());
+
+		OGR_G_SetPoint_2D(pt.get(), 0, proj_cord.x(), proj_cord.y());
+
+		if (need_wgs1984)
+			OGR_G_Transform(pt.get(), transformation.get());
+
+		OGR_F_SetGeometry(po_feature.get(), pt.get());
+
+		OGR_F_SetStyleString(po_feature.get(),
+		                     OGR_STBL_Find(table.get(), QString::number(map->findSymbolIndex(object->getSymbol())).toLatin1().constData()));
+
+		if (OGR_L_CreateFeature(layer, po_feature.get()) != OGRERR_NONE)
+			throw FileFormatException(tr("Failed to create point feature in layer"));
+	};
+
+	map->applyOnMatchingObjects(add_feature, condition);
+}
+
+void OgrFileExport::addLinesToLayer(OGRLayerH layer, const std::function<bool (const Object*)>& condition)
+{
+	const auto& georef = map->getGeoreferencing();
+
+	auto add_feature = [&](Object* object) {
+		auto symbol = object->getSymbol();
+		auto path = object->asPath();
+		if (path->parts().size() < 1)
+			return;
+
+		auto po_feature = ogr::unique_feature(OGR_F_Create(OGR_L_GetLayerDefn(layer)));
+
+		QString sym_name = symbol->getPlainTextName();
+		sym_name.truncate(32);
+		OGR_F_SetFieldString(po_feature.get(), OGR_F_GetFieldIndex(po_feature.get(), "Name"), sym_name.toLatin1().constData());
+
+		auto line_string = ogr::unique_geometry(OGR_G_CreateGeometry(wkbLineString));
+		auto num_coords = path->getCoordinateCount();
+		for (MapCoordVector::size_type i = 0; i < num_coords; i++)
+		{
+			// Linear approx of bezier curve
+			if (path->getCoordinate(i).isCurveStart())
+			{
+				for (double j = 0; j <= 10.0; j++)
+				{
```

Comment:
I've made each bezier curve into 10 sections - should this be dynamic based on the distance or a smaller/larger number?